tests/test_variables.py

```diff
@@ -43,15 +39,70 @@ def test_count(self):
                 self.assertEqual(variables.count(v), 0)
 
     def test_len(self):
-        variables = Variables(range(5))
-        self.assertEqual(len(variables), 5)
         variables = Variables('aaaaa')
         self.assertEqual(len(variables), 1)
 
+
+class TestList(unittest.TestCase):
+    iterable = list(range(5))
+
+    def test_index_api(self):
+        variables = Variables(self.iterable)
+        self.assertTrue(hasattr(variables, 'index'))
+        self.assertTrue(callable(variables.index))
+        self.assertTrue(isinstance(variables.index, abc.Mapping))
+
     def test_contains_unhashable(self):
-        variables = Variables(range(5))
-        self.assertFalse([0] in variables)
+        variables = Variables(self.iterable)
+        self.assertFalse([] in variables)
 
     def test_count_unhashable(self):
-        variables = Variables(range(5))
-        self.assertEqual(variables.count([5]), 0)
+        variables = Variables(self.iterable)
+        self.assertEqual(variables.count([]), 0)
+
+    def test_index(self):
+        variables = Variables(self.iterable)
+        for idx, v in enumerate(self.iterable):
+            self.assertEqual(variables.index(v), idx)
+
+    def test_iterable(self):
+        variables = Variables(self.iterable)
+        self.assertEqual(list(variables), list(self.iterable))
+
+    def test_equality(self):
+        variables = Variables(self.iterable)
+        self.assertEqual(variables, self.iterable)
+
+    def test_len(self):
+        variables = Variables(self.iterable)
+        self.assertEqual(len(variables), len(self.iterable))
+
+    def test_relabel_conflict(self):
+        variables = Variables(self.iterable)
+
+        iterable = self.iterable
+        target = [iterable[-i] for i in range(len(iterable))]
```

Comment:
Do you have to use the partial reversal (i.e. index is [0,4,3,2,1]), or is a full reversal okay (i.e. index is [4,3,2,1,0])?

If full reversal is okay, perhaps consider `target = iterable[::-1]`

Partial reversal could also be `target = iterable[0] + iterable[:0:-1]`, but it may not look as nice.

Reply:
Really I just wanted a relabelling that mapped the labels to themselves with some identity relabels. Agree with `[:,:,-1]` syntax is nicer.

Reply:
I have added a comment making the intention clearer